addons/hr_holidays/tests/test_accrual_allocations.py

```diff
@@ -1340,3 +1346,38 @@ def test_accrual_carrover_other_period_end_multi_level(self):
         with freeze_time("2026-08-01"):
             allocation._update_accrual()
         self.assertAlmostEqual(allocation.number_of_days, 1.315, 3)
+
+    def test_accrual_creation_on_anterior_date(self):
+        accrual_plan = self.env['hr.leave.accrual.plan'].with_context(tracking_disable=True).create({
+            'name': 'Weekly accrual',
+            'carryover_date': 'allocation',
+            'level_ids': [(0, 0, {
+                'added_value_type': 'day',
+                'start_count': 0,
+                'start_type': 'day',
+                'added_value': 1,
+                'frequency': 'weekly',
+                'cap_accrued_time': False,
+                'action_with_unused_accruals': 'lost',
+            })],
+        })
+        with freeze_time('2023-09-01'):
+            accrual_allocation = self.env['hr.leave.allocation'].new({
+                'name': 'Employee allocation',
+                'holiday_status_id': self.leave_type.id,
+                'date_from': '2023-01-01',
+                'employee_id': self.employee_emp.id,
+                'allocation_type': 'accrual',
+                'accrual_plan_id': accrual_plan.id,
+            })
+            # As the duration is set to a onchange, we need to force that onchange to run
+            accrual_allocation._onchange_date_from()
```

Comment:
Indeed, that was my first idea, but I think it's more explicit as to how the duration allocated is set that way :thinking: Though in the first place I find a onchange rather than a compute rather strange... What do you think?

Reply:
To my knowledge, this was only done for a form view. So that you can have a feeling of how much days you get with a certain accruel. So, onchange does a trick 
Maybe we will need to change in the future.. but for current needs, ca va